### ISSUE TITLE
docs: Fit models with standard for loop + notebook on training loops.

### ISSUE THREAD

**Issue by daniel-dodd** [JaxGaussianProcesses/GPJax#382]
We have a convenient `fit` function to train GPs against objectives. It would be good though e.g., in the regression notebook to show a simple python for loop, and a simple `lax.scan` training loop, to demonstrate that users can write their own training loops. 

Then it would be good to link this to a more extensive notebook exposing users to stoping gradients, bijectors transformations etc, and show how to add a training bar to the loop.
